bolt-sidecar/bin/sidecar.rs

```diff
@@ -1,140 +1,167 @@
 
                 tracing::info!(
-                    tx_hash = %request.tx.tx_hash(),
+                    tx_hash = %request.tx.hash(),
                     "Validation against execution state passed"
                 );
 
                 // parse the request into constraints and sign them with the sidecar signer
                 // TODO: get the validator index from somewhere
-                let message = ConstraintsMessage::build(0, request.slot, request.clone());
+                let message = ConstraintsMessage::build(0, request.slot, request);
 
-                let signature = signer.sign(&message.digest())?;
-                let signed_constraints: BatchedSignedConstraints =
-                    vec![SignedConstraints { message, signature: signature.to_string() }];
+                let signature = signer.sign(&message.digest())?.to_string();
+                let signed_constraints = vec![SignedConstraints { message, signature }];
 
                 // TODO: fix retry logic
                 let max_retries = 5;
                 let mut i = 0;
-                while let Err(e) = mevboost_client
+                'inner: while let Err(e) = mevboost_client
                     .submit_constraints(&signed_constraints)
                     .await
                 {
-                    tracing::error!(error = ?e, "Error submitting constraints, retrying...");
+                    tracing::error!(err = ?e, "Error submitting constraints, retrying...");
                     tokio::time::sleep(Duration::from_millis(100)).await;
                     i+=1;
                     if i >= max_retries {
-                        break
+                        break 'inner
                     }
                 }
-            }
-            Some(request) = payload_rx.recv() => {
-                tracing::info!("Received local payload request: {:?}", request);
-                let Some(response) = execution_state.get_block_template(request.slot) else {
-                    tracing::warn!("No block template found for slot {} when requested", request.slot);
-                    let _ = request.response.send(None);
+
+                let res = serde_json::to_value(signed_constraints).map_err(Into::into);
+                let _ = response_tx.send(res).ok();
+            },
+            Ok(HeadEvent { slot, .. }) = head_tracker.next_head() => {
+                tracing::info!(slot, "Received new head event");
+
+                // We use None to signal that we want to fetch the latest EL head
+                if let Err(e) = execution_state.update_head(None).await {
+                    tracing::error!(err = ?e, "Failed to update execution state head");
+                }
+
+                if let Err(e) = consensus_state.update_head(slot).await {
+                    tracing::error!(err = ?e, "Failed to update consensus state head");
+                }
+            },
+            Some(slot) = consensus_state.commitment_deadline.wait() => {
```

Comment:
Question: If this fires, is it ever reset?

Reply:
it's reset here: https://github.com/chainbound/bolt/pull/97/files#diff-c56124d587df110d574887395349f93b1180ac9dcc466cb5201faaf104c817beR102-R104